nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -766,6 +776,23 @@ in
         '';
       };
 
+      addMimeTypes = mkOption {
+        type = types.submodule {
+          freeformType = types.attrsOf (types.listOf types.str);
+        };
+        default = { };
+        example = literalExpression ''
+          "application/atom+xml" = [ "atom" ];
+          "image/jpeg" = [ "jpg" "jpef" ];
+          "text/css" = [ "css" ];
+        '';
+        description = ''
+          Enabling custom mime types to an existing list {option}`services.nginx.defaultMimeTypes`.
+          This option does not allow overriding existing mime-types in
+          {option}`services.nginx.defaultMimeTypes` and may cause duplicate mime-types.
```

Comment:
The description needs an update: this behavior can be changed by setting defaultMimeTypes to null.